src/consensus/chainsel.md

```diff
@@ -0,0 +1,259 @@
+# Chain selection
+
+_Chain selection_ is the mechanism, specified in the Protocol associated to the
+current era in effect in the network, of identifying and and adopting the best
+chain in the network. A node might receive distinct chains from its peers. This
+situation is expected during normal operation of the Praos protocol and its
+derivatives.
+
+It is important to note that **all nodes** participating in the network are in
+charge of this responsibility, as opposed to producing blocks, which only some
+nodes will be involved in doing (namely those with forging credentials, the
+cryptographic keys for signing new blocks on behalf of a stake pool). This
+implies that even non-block-forging nodes contribute to the security of the
+network as a whole.
+
+Because each node performs chain selection, the Cardano network acts as a
+forward-filtering diffusion network.
+
+The block headers come from [`ChainSync`](./miniprotocols/chainsync.md) and
+the block bodies come from [`BlockFetch`](./miniprotocols/blockfetch.md).
+
+> [!NOTE]
+>
+> Chain selection is performed based on data from the previous headers, so there
+> is some state (specific to each protocol) that influences the selection, which
+> we will call _chain state_, as opposed to the state of the ledger or
+> _on-chain_ state, which we will call _ledger state_.
+
+It is the decisions from running _chain selection_ the ones that mutate the data
+in [the Storage layer](../storage). Also the newly selected chain will influence
+[the Mempool](../mempool) as transactions will be revalidated against this new
+chain.
+
+<!-- toc -->
+
+## Validating headers
+
+The validity of the headers of a candidate can be checked using the Chain State
+and Ledger State at the intersection point, as long as the distance in slots
+between the intersection and the candidate header is no more than the forecast
+range. For example, in this chains, the headers of the blocks in the candidate
+that are no further than the forecast range in slots from block 1 could be
+validated.
+
+```mermaid
+---
+config:
+  gitGraph:
+    mainBranchName: "selected chain"
+---
+gitGraph
+    commit
+    commit
+    branch candidate
+    checkout candidate
+    commit
+    commit
+    checkout "selected chain"
+    commit
+    checkout candidate
+    commit
+    commit
+    checkout "selected chain"
+    commit
+```
+
+In Praos, the forecast range is set to `3k/f = 129,600` slots. This is possible
```

Comment:
I don't get why the second sentence follows from the first :thinking:

Reply:
Aha, the sentence belongs before the graph. Good catch